Branch-SDK/Branch-SDK/BranchEvent.h

```diff
@@ -0,0 +1,75 @@
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventAddPaymentInfo;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventPurchase;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventSpendCredits;
+
+///@name Content Events
+
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventSearch;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventViewItem;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventViewItems;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventRate;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventShare;
+
+///@name User Lifecycle Events
+
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventCompleteRegistration;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventCompleteTutorial;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventAchieveLevel;
+FOUNDATION_EXPORT BranchStandardEvent _Nonnull BranchStandardEventUnlockAchievement;
+
+#pragma mark - BranchEvent
+
+@interface BranchEvent : NSObject
+
+- (instancetype _Nonnull) initWithName:(NSString*_Nonnull)name NS_DESIGNATED_INITIALIZER;
+
++ (instancetype _Nonnull) standardEvent:(BranchStandardEvent _Nonnull)standardEvent;
++ (instancetype _Nonnull) standardEvent:(BranchStandardEvent _Nonnull)standardEvent
+                        withContentItem:(BranchUniversalObject* _Nonnull)contentItem;
+
++ (instancetype _Nonnull) customEventWithName:(NSString*_Nonnull)name;
++ (instancetype _Nonnull) customEventWithName:(NSString*_Nonnull)name
+                                  contentItem:(BranchUniversalObject*_Nonnull)contentItem;
+
+- (instancetype _Nonnull) init __attribute((unavailable));
++ (instancetype _Nonnull) new __attribute((unavailable));
+
+@property (nonatomic, strong) NSString*_Nullable                transactionID;
+@property (nonatomic, strong) BNCCurrency _Nullable             currency;
+@property (nonatomic, strong) NSDecimalNumber*_Nullable         revenue;
+@property (nonatomic, strong) NSDecimalNumber*_Nullable         shipping;
+@property (nonatomic, strong) NSDecimalNumber*_Nullable         tax;
+@property (nonatomic, strong) NSString*_Nullable                coupon;
+@property (nonatomic, strong) NSString*_Nullable                affiliation;
+@property (nonatomic, strong) NSString*_Nullable                eventDescription;
+@property (nonatomic, strong) NSString*_Nullable                searchQuery;
+@property (nonatomic, copy) NSMutableArray<BranchUniversalObject*>*_Nonnull       contentItems;
+@property (nonatomic, copy) NSMutableDictionary<NSString*, NSString*> *_Nonnull   customData;
+
+- (void) logEvent;                      //!< Logs the event on the Branch server.
+- (NSDictionary*_Nonnull) dictionary;   //!< Returns a dictionary representation of the event.
+- (NSString* _Nonnull) description;     //!< Returns a string description of the event.
```

Comment:
what's the difference between description and eventDescription?

Reply:
Since `description` is a reserved word in obj-c I renamed that field `eventDescription`.

The `description` method returns a description on the whole class.

Reply:
ok. as long as we're sending up "description" to the server, i suppose that's fine